packages/amplify_core/lib/src/types/models/auth_rule.dart

```diff
@@ -22,7 +22,17 @@ import 'package:flutter/foundation.dart';
 
 enum AuthStrategy { OWNER, GROUPS, PRIVATE, PUBLIC }
 
-enum ModelOperation { CREATE, UPDATE, DELETE, READ }
+enum ModelOperation {
+  CREATE,
+  UPDATE,
+  DELETE,
+  READ,
+  GET,
+  LIST,
+  SYNC,
```

Comment:
Per my understanding these finer read operation controls work out of box for API plugin, are these finer controls supported in DataStore? If not, do we have a documentation calling this out - not to use these finer read control with DataStore?

Reply:
These are not supported in Datastore especially for the local storage. The auth rules are only respected when it syncs with cloud. Adding these enums is to make the modelgen file compiled for those who uses these files only for API usage.

There is no document on this part but I will have a PR out later to call out that the finer read ops are not recommended for Datastore usage. If cx insists on using them, the `sync` and `listen` are required.

Reply:
Update with the doc PR: https://github.com/aws-amplify/docs/pull/5084